src/west/commands/project.py

```diff
@@ -999,3 +1018,23 @@ def canonical(path):
 MANIFEST_URL_DEFAULT = 'https://github.com/zephyrproject-rtos/zephyr'
 # Default revision to check out of the manifest repository.
 MANIFEST_REV_DEFAULT = 'master'
+
+#
+# Helper class for creating format string keys that are expensive or
+# undesirable to compute if not needed.
+#
+
+class DelayFormat:
+
+    def __init__(self, obj):
+        self.obj = obj
+        self.as_str = None
+
+    def __format__(self, format_spec):
+        if self.as_str is None:
+            if callable(self.obj):
```

Comment:
Could you add a comment to the class description that states that if the object an instance is constructed with is a function (I assume that you didn't really intend for a class instance that is callable to actually be called here) then it will be called instead of stringyfied with `str()`?

Reply:
Done; ready for review.

Reply:
>  Done; ready for review.

Whoops. Need to remove spaces from the {cloned} format.

Reply:
OK, this time for real.